prow/config/config.go

```diff
@@ -1092,10 +1098,30 @@ type Deck struct {
 	Branding *Branding `json:"branding,omitempty"`
 	// GoogleAnalytics, if specified, include a Google Analytics tracking code on each page.
 	GoogleAnalytics string `json:"google_analytics,omitempty"`
+	// RerunAuthConfigs is depricated in favor of the slice structure DefaultRerunAuthConfigs.
+	// It remains a part of Deck for the purposes of backwards compatibility.
 	// RerunAuthConfigs is a map of configs that specify who is able to trigger job reruns. The field
 	// accepts a key of: `org/repo`, `org` or `*` (wildcard) to define what GitHub org (or repo) a particular
 	// config applies to and a value of: `RerunAuthConfig` struct to define the users/groups authorized to rerun jobs.
 	RerunAuthConfigs RerunAuthConfigs `json:"rerun_auth_configs,omitempty"`
+	// DefaultRerunAuthConfigs is a list of DefaultRerunAuthConfigEntry structures that specify who can
+	// trigger job reruns. Reruns are based on whether the entry's org/repo or cluster matches with the
+	// expected fields in the given configuration.
+	//
+	// This field is mutually exclusive with the RerunAuthConfigs field.
+	DefaultRerunAuthConfigs DefaultRerunAuthConfigs `json:"default_rerun_auth_configs,omitempty"`
+	// DefaultRerunAuthConfigEntries is used to populate DefaultRerunAuthConfigs.
+	//
+	// Each entry in the slice specifies Repo and Cluster regexp filter fields to
+	// match against jobs and a corresponding RerunAuthConfig. The entry matching the job with the
+	// most specification is for authentication purposes.
+	//
+	// This field is smarter than the RerunAuthConfigs, because each
+	// entry includes additional Cluster regexp information that the old format
+	// does not consider.
+	//
+	// This field is mutually exclusive with the DefaultDecorationConfigsMap field.
+	DefaultRerunAuthConfigEntries DefaultRerunAuthConfigs `json:"default_rerun_auth_config_entries,omitempty"`
```

Comment:
this is duplicated with `DefaultRerunAuthConfigs` above, my feeling is that only one of them is needed, is there anything that I'm missing?